File: src/main/java/cc/hyperium/internal/addons/OverlayChecker.java
package cc.hyperium.internal.addons;

import cc.hyperium.gui.main.HyperiumOverlay;
import cc.hyperium.internal.addons.misc.AddonLoadException;

public class OverlayChecker {
    public static void checkOverlayField(String value) throws AddonLoadException {
        try {
            Class<?> originClass = Class.forName(value);
            if (HyperiumOverlay.class.isInstance(originClass)) {
                throw new AddonLoadException("overlay has to be an instance of HyperiumOverlay");
            }
        } catch (ClassNotFoundException e) {
            throw new AddonLoadException("Overlay class doesn't exist");
        }
    }
}

File: src/main/java/cc/hyperium/internal/addons/OverlayChecker.kt
package cc.hyperium.internal.addons

import cc.hyperium.gui.main.HyperiumOverlay
import cc.hyperium.internal.addons.misc.AddonLoadException

object OverlayChecker {
    @Throws(AddonLoadException::class)
    fun checkOverlayField(value: String) {
        val originClass = Class.forName(value)
        if (HyperiumOverlay::class.java.isInstance(originClass)) {
            throw AddonLoadException("overlay has to be an instance of HyperiumOverlay")
        }
    }
}
